### PR TITLE
17 generation of documentation

### PR DESCRIPTION
**Pull request by christophechr**
## Description
Ajout d'une documentation API locale via swagger

## Changement
Ajout de documentation API

## Commentaire
N/A

### CODE REVIEW

File: backend/playfit/authentification/views.py [hunk 1]
```diff
@@ -23,6 +33,10 @@ def post(self, request):
 class LoginView(APIView):
     permission_classes = [AllowAny]
 
+    @swagger_auto_schema(
+        operation_description="Tentative de connexion d'un utilisateur",
```
Comment (Eowiin) [inline backend/playfit/authentification/views.py:37]: Update description in English, add request body and all the responses possible. Same for logout view

Comment (Masteruuuuu) [inline backend/playfit/authentification/views.py:37]: Just change to english and all seems good

**Review by Eowiin — APPROVED** [review]

File: backend/playfit/authentification/views.py [hunk 2]
```diff
@@ -6,10 +6,29 @@
...
     permission_classes = [AllowAny]
 
+    @swagger_auto_schema(
+        request_body=CustomUserSerializer,
+        responses={
+            201: openapi.Response("User registered successfully", CustomUserSerializer),
+            400: "Invalid data",
+        },
+        operation_description="Register a new user with username, email, height, weight, date of birth, and password.",
+        request_body=openapi.Schema(
```
Comment (Eowiin) [inline backend/playfit/authentification/views.py:22]: Remove this request body or the previous one in this schema for this view

File: backend/playfit/authentification/views.py [hunk 3]
```diff
@@ -23,6 +42,22 @@ def post(self, request):
 class LoginView(APIView):
     permission_classes = [AllowAny]
 
+    @swagger_auto_schema(
+        operation_description="User login with username or email and password.",
+        responses={
+            200: openapi.Response("User logged in successfully"),
+            400: "Invalid credentials",
+        }
```
Comment (Eowiin) [inline backend/playfit/authentification/views.py:50]: Missing comma at the end of the line

**Review by Eowiin — CHANGES_REQUESTED** [review]

**Review by Eowiin — APPROVED** [review]

**Pull request merged**